shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -296,13 +363,127 @@ private static Integer inputActionFromTextInputAction(@NonNull String inputActio
       }
     }
 
+    public static class Autofill {
+      public static Autofill fromJson(@NonNull JSONObject json)
+          throws JSONException, NoSuchFieldException {
+        final String uniqueIdentifier = json.getString("uniqueIdentifier");
+        final JSONArray hints = json.getJSONArray("hints");
+        final JSONObject editingState = json.getJSONObject("editingValue");
+        final String[] hintList = new String[hints.length()];
+
+        for (int i = 0; i < hintList.length; i++) {
+          hintList[i] = translateAutofillHint(hints.getString(i));
+        }
+        return new Autofill(uniqueIdentifier, hintList, TextEditState.fromJson(editingState));
+      }
+
+      public final String uniqueIdentifier;
+      public final String[] hints;
+      public final TextEditState editState;
+
+      @NonNull
+      private static String translateAutofillHint(@NonNull String hint) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
nit: it might be a bit more readable if you do:

```java
if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
  return hint;
}
// then switch 
```